### DISCUSSION TITLE
Change language url

### DISCUSSION THREAD

**Discussion by Smiog** [razonyang/hugo-theme-bootstrap#360] [Q&A] [answered]
How can I change a language's url? For example, I want to change 'en' into 'en-us', and I can access my site in this language in .../en-us but not .../en.

**Comment by razonyang** [comment, marked as answer]
1. Change `[en]` to `[en-us]`.
https://github.com/razonyang/hugo-theme-bootstrap/blob/47729f6eb1752ff7f5a164ed717f2e62b738f254/exampleSite/config/_default/languages.toml#L1

2. Change the default language to `en-us` if necessary.
https://github.com/razonyang/hugo-theme-bootstrap/blob/47729f6eb1752ff7f5a164ed717f2e62b738f254/exampleSite/config/_default/config.toml#L7

3. Modify the content filename or folder name if necessary, for example: `index.en.md` to `index.en-us.md`, `content/en` to `content/en-us`.

**Reply by Smiog** [reply]
I can change the url, but after I modified the filename, hugo server can't run.
![image](https://user-images.githubusercontent.com/90970249/151759797-ae9d331d-de5c-4af8-b712-7534ef94b740.png)
Here's the error, how can I solve it?

**Reply by razonyang** [reply]
I guess you're changing the `zh-cn` to `zh`. If so, you'll need to change all filenames from `*.zh-cn.*` to `*.zh.*`. For example:

- `docs/widgets/comments/index.zh-cn.md` to  `docs/widgets/comments/index.zh.md`, so that `ref` can find the page writing in same language.

**Comment by Smiog** [comment]
If I don't wanna keep these posts, how can I delete them?

**Reply by razonyang** [reply]
The example site content is used to show common use cases(config, params and menus etc). You can remove them by deleting the `content/docs` and `content/posts` folders. There are also configurations and parameters need to be clean up, such as `config.zh-cn.toml`, `params.zh-cn.toml`.

**Comment by Smiog** [comment]
What's more, 
![image](https://user-images.githubusercontent.com/90970249/151909421-ed44653c-49f2-44a3-8548-b97f6fcbd247.png)
![image](https://user-images.githubusercontent.com/90970249/151909438-b06eb82a-75c8-41c8-9e1e-10da91cd0b17.png)
and,
![image](https://user-images.githubusercontent.com/90970249/151909463-4c0f5367-c1ad-477c-9b23-7b7b21dc1657.png)
![image](https://user-images.githubusercontent.com/90970249/151909491-fb63bc1f-555f-4da3-948d-bbcb8196e97e.png)
besides,
![image](https://user-images.githubusercontent.com/90970249/151909613-59ad9f96-d812-42e4-bf06-c68dc63e58f6.png)
![image](https://user-images.githubusercontent.com/90970249/151909630-b96f502e-9962-4759-84a9-f99cbbebe57f.png)
and so on...
Why are there so many bugs after changing?

**Reply by razonyang** [reply]
It's not a bug, you need to copy the [`zh-cn.toml`](https://github.com/razonyang/hugo-theme-bootstrap/blob/master/i18n/zh-cn.toml) to `i18n/zh.toml`(translations, depends on the language code). The I18N will fallback to the default language if expected `i18n/{lang}.toml` is not found.

If you're changing the language code, you need to change all relevant configuration(menus, parameters) and posts.

I also suggest reading further on [Hugo official docs](https://gohugo.io/getting-started/).
